File: Nx3CS/bin/test/src/nx3/PHead.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PHead : global::haxe.lang.HxObject {
		public    PHead(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PHead(global::nx3.NHead nhead){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\PHead.hx"
				global::nx3.PHead.__hx_ctor_nx3_PHead(this, nhead);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PHead(global::nx3.PHead __temp_me100, global::nx3.NHead nhead){
			unchecked {
				#line 12 "F:\\nx3\\src\\nx3\\PHead.hx"
				__temp_me100.nhead = nhead;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				return new global::nx3.PHead(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				return new global::nx3.PHead(((global::nx3.NHead) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.NHead nhead;
		
		public  global::nx3.PNote note;
		
		public virtual   global::nx3.PNote getNote(){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\PHead.hx"
				return this.note;
			}
			#line default
		}
		
		
		public virtual   string toString(){
			unchecked {
				#line 21 "F:\\nx3\\src\\nx3\\PHead.hx"
				return global::haxe.lang.Runtime.concat("PHead  \r", global::Std.@string(this.nhead));
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				switch (hash){
					case 1225398258:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						this.note = ((global::nx3.PNote) (@value) );
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return @value;
					}
					
					
					case 455014542:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						this.nhead = ((global::nx3.NHead) (@value) );
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return @value;
					}
					
					
					default:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				switch (hash){
					case 946786476:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("toString") ), ((int) (946786476) ))) );
					}
					
					
					case 462210984:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getNote") ), ((int) (462210984) ))) );
					}
					
					
					case 1225398258:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return this.note;
					}
					
					
					case 455014542:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return this.nhead;
					}
					
					
					default:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				switch (hash){
					case 946786476:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return this.toString();
					}
					
					
					case 462210984:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return this.getNote();
					}
					
					
					default:
					{
						#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				baseArr.push("note");
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				baseArr.push("nhead");
				#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
				{
					#line 7 "F:\\nx3\\src\\nx3\\PHead.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
		public override string ToString(){
			return this.toString();
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PHead.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PHead extends haxe.lang.HxObject
{
	public    PHead(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PHead(nx3.NHead nhead)
	{
		nx3.PHead.__hx_ctor_nx3_PHead(this, nhead);
	}
	
	
	public static   void __hx_ctor_nx3_PHead(nx3.PHead __temp_me117, nx3.NHead nhead)
	{
		__temp_me117.nhead = nhead;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PHead(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PHead(((nx3.NHead) (arr.__get(0)) ));
	}
	
	
	public  nx3.NHead nhead;
	
	public  nx3.PNote note;
	
	public   nx3.PNote getNote()
	{
		return this.note;
	}
	
	
	@Override public   java.lang.String toString()
	{
		return ( "PHead  \r" + haxe.root.Std.string(this.nhead) );
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef701 = true;
			switch (field.hashCode())
			{
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef701 = false;
						this.note = ((nx3.PNote) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 104785742:
				{
					if (field.equals("nhead")) 
					{
						__temp_executeDef701 = false;
						this.nhead = ((nx3.NHead) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef701) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef702 = true;
			switch (field.hashCode())
			{
				case -1776922004:
				{
					if (field.equals("toString")) 
					{
						__temp_executeDef702 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("toString"))) );
					}
					
					break;
				}
				
				
				case 104785742:
				{
					if (field.equals("nhead")) 
					{
						__temp_executeDef702 = false;
						return this.nhead;
					}
					
					break;
				}
				
				
				case -75294616:
				{
					if (field.equals("getNote")) 
					{
						__temp_executeDef702 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getNote"))) );
					}
					
					break;
				}
				
				
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef702 = false;
						return this.note;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef702) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef703 = true;
			switch (field.hashCode())
			{
				case -1776922004:
				{
					if (field.equals("toString")) 
					{
						__temp_executeDef703 = false;
						return this.toString();
					}
					
					break;
				}
				
				
				case -75294616:
				{
					if (field.equals("getNote")) 
					{
						__temp_executeDef703 = false;
						return this.getNote();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef703) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("note");
		baseArr.push("nhead");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


